### ISSUE TITLE
Cancelling bid has a few issues

### ISSUE THREAD

**Issue by Anonomit** [Vapok/CommunityDKP#100]
Cancelling a bid will silently remove the player from the list of bidders, and will not go through the same logic as successfully adding a bid. It will not activate the antisnipe timer, and will not cause the current bid to be broadcast.

![image](https://user-images.githubusercontent.com/67178487/87268877-96f68480-c499-11ea-9d78-cb845f530c47.png)
![image](https://user-images.githubusercontent.com/67178487/87269126-403d7a80-c49a-11ea-905b-5a5d56c2ce97.png)

1. Start a bid.
2. Bid any amount.
3. Some else bid a higher amount and then cancel their bid (preferably within the antisnipe window).
4. Notice that the bid timer was not extended, and the player running the bid has not announced the new highest bid.

In this situation, the bid timer should be extended and the new highest bid should be announced by the player running the bid.

r3.1.2-r44

**Comment by f90** [comment]
I would argue this is a very high priority bug to fix since it leaves open bidding VERY vulnerable to bid manipulation, and is barely usable. This issue was already there in Monolith by the way, but got never fixed. 

Having the option to not allow cancelling bids might also be a useful addition that is related to this.

**Comment by Vapok** [comment]
@f90 @Anonomit 

How's this for a possible solution:
When the timer reaches the 10 second mark, if AntiSnipe is activated, it will not allow bid cancels.

Does that solve the immediate actual issue?

**Comment by sarahbau** [comment]
I don't think disallowing bid cancelling in the last 10 seconds should be the default. I could see that being something some people might want an option for though. For my raid I'd much rather it extend the time and announce the new high the same way a new bid would.

**Comment by f90** [comment]
@Vapok I don't think this solves the issue since you still have the problem of people bidding high, then cancelling early, and no one is informed about the new highest bid (which is now lower) - so people will miss the opportunity of bidding on an item since they think it's too expensive for them.

Besides I dont expect the implementation to be that difficult? Just check on cancellation if that changes the highest bid, if yes, then trigger the usual announcement + extend bid timer if anti-snipe is active?

**Comment by Vapok** [comment]
Okay, I'll keep looking at a solution. The cancellation is a little more complicated, but I'll take another pass at it.
